### PR TITLE
[PORT] Port MIS Builder in 9.0

### PR DESCRIPTION
**Pull request by adrienpeiffer**
Supersedes #147

### CODE REVIEW

File: mis_builder/models/aep.py [hunk 1]
```diff
@@ -87,55 +66,33 @@ def __init__(self, env):
...
                 continue
             if account_code is None:
-                # by convention the root account is keyed as
-                # None in _account_ids_by_code, so it is consistent
-                # with what _parse_match_object returns for an
-                # empty list of account codes, ie [None]
-                exact_codes.add(root_account.code)
+                # None means we want all accounts
+                account_ids = account_model.\
+                    search([]).mapped('id')
```
Comment (sebastienbeau) [inline mis_builder/models/aep.py:84]: Hi you can use account_model.search().ids

File: mis_builder/models/aep.py [hunk 2]
```diff
@@ -87,55 +66,33 @@ def __init__(self, env):
...
+                self._account_ids_by_code[account_code].update(account_ids)
             elif '%' in account_code:
-                like_codes.add(account_code)
+                account_ids = account_model.\
+                    search([('code', 'like', account_code)]).mapped('id')
```
Comment (sebastienbeau) [inline mis_builder/models/aep.py]: same here

**Comment by sebastienbeau** [comment]
code review, minor comment :+1:

File: mis_builder/static/src/js/mis_builder.js [hunk 3]
```diff
@@ -1,48 +1,151 @@
...
+    template: "mis_builder.MisReport",
+    init: function() {
+        this._super.apply(this, arguments);
+        this.mis_report_data = null;
+        this.mis_report_instance_id = false;
+        this.field_manager.on("view_content_has_changed", this, this.reload_widget);
+    },
+    
+    reload_widget: function() {
+        var self = this
```
Comment (tarteo) [inline mis_builder/static/src/js/mis_builder.js:25]: Missing ;

File: mis_builder/static/src/js/mis_builder.js [hunk 4]
```diff
@@ -1,48 +1,151 @@
...
+        self.mis_report_instance_id = self.getParent().datarecord.id
```
Comment (tarteo) [inline mis_builder/static/src/js/mis_builder.js:26]: Missing ;

File: mis_builder/static/src/js/mis_builder.js [hunk 5]
```diff
@@ -1,48 +1,151 @@
...
+            self.getParent().dataset.context['no_destroy'] = true;
+            self.generate_content();
+        } else {
+            self.display_settings();
+        }
+    },
+    
+    get_context: function() {
+        var self = this;
+        var context = {}
```
Comment (tarteo) [inline mis_builder/static/src/js/mis_builder.js:48]: Missing ;

File: mis_builder/static/src/js/mis_builder.js [hunk 6]
```diff
@@ -1,48 +1,151 @@
...
+        if (this.mis_report_instance_id){
+            context['active_ids'] = [this.mis_report_instance_id];
+        }
+        return context
+    },
+    
+    print: function() {
+        var self = this
```
Comment (tarteo) [inline mis_builder/static/src/js/mis_builder.js:56]: Missing ;

File: mis_builder/static/src/js/mis_builder.js [hunk 7]
```diff
@@ -1,48 +1,151 @@
...
+        var context = new data.CompoundContext(self.build_context(), self.get_context()|| {})
```
Comment (tarteo) [inline mis_builder/static/src/js/mis_builder.js:57]: Missing ;

**Comment by sbidoul** [comment]
Superseded by #189

**Pull request closed without merging**